### PR TITLE
Fix broken zlib unittests

### PR DESCRIPTION
**Pull request by armenatzoglou**
Previous commit broke two unit-tests on workfile's compression using zlib. The broken unit-tests assume that g_datasourceCtx is allocated under TopMemoryContext, but in the previous commit we changed that to TopTransactionContext. This PR updates the unit-tests accordingly.

@foyzur @gcaragea Please take a look.

### CODE REVIEW

**Comment by gcaragea** [comment]
:+1: 
Thanks for fixing this.

File: src/backend/storage/file/test/compress_zlib_test.c [hunk 1]
```diff
@@ -80,11 +80,11 @@ test__bfz_zlib_init__palloc_write(void **state)
 	bfz.mode = BFZ_MODE_APPEND;
 	bfz.fd = -1;
 
-	Size beforeAlloc = MemoryContextGetPeakSpace(TopMemoryContext);
+	Size beforeAlloc = MemoryContextGetPeakSpace(TopTransactionContext);
```
Comment (foyzur) [inline src/backend/storage/file/test/compress_zlib_test.c]: Can we add an assert that the current context is NOT TopTransactionContext?

**Comment by foyzur** [comment]
LGTM except the mentioned assert. We should check if TopTransactionContext automatically becomes active AND increases balance.

**Comment by armenatzoglou** [comment]
@foyzur Good point! Thanks. I added an assert after each MemoryContextGetPeakSpace(TopTransactionContext) call.

**Comment by armenatzoglou** [comment]
Pushed (https://github.com/greenplum-db/gpdb/commit/3d22af6253d31cffd2171ccfbd5bc321279f8716). Thanks a lot for the review.

**Pull request closed without merging**